Sources/Yams/Constructor.swift

```diff
@@ -360,13 +360,34 @@ private extension String {
     }
 }
 
+#if swift(>=4.1)
+
+private extension StringProtocol where Self.SubSequence == Substring {
```

Comment:
Any reason why this can't be

```swift
private extension StringProtocol {
    func substring(from offset: IndexDistance) -> SubSequence {
        return self[index(startIndex, offsetBy: offset)...]
    }
}
```

It looks like it compiles going back to Swift 4.0.0 on both Linux & Darwin, so we wouldn't need the explicit `String` and `Substring` versions below.